lib/salad_ui/helpers.ex

```diff
@@ -9,8 +9,8 @@ defmodule SaladUI.Helpers do
     assigns
     |> assign(field: nil, id: assigns[:id] || field.id)
     # |> assign(:errors, Enum.map(field.errors, &translate_error(&1)))
-    |> assign(:name, if(assigns[:multiple], do: field.name <> "[]", else: field.name))
-    |> assign(:value, field.value)
+    |> assign_new(:name, fn -> if(assigns[:multiple], do: field.name <> "[]", else: field.name) end)
+    |> assign_new(:value, fn -> field.value end)
```

Comment:
This change will cause a huge impact on all input component. Because when you define an attribute without default value, and you don't pass that attribute to the component, then it will raise an exception when accessing that attribute within that component.
Example:

```elixir
attr :name, :string

def input(assigns) do
~H"""
<input name={@name} />
"""
end
```

this is okay
```elixir
<.input name="title"/>
```

but this raise exception
```elixir
<.input id="my-input"/>
```

I think it needs a big refactor to be able to override name and value

Reply:
the error occur when user doesn't set name and field, in this case, access `@name` cause exception

Reply:
I think we can handle this way

```elixir
...
|> assign(:name, assign[:name] || if(assigns[:multiple], do: field.name <> "[]", else: field.name))
|> assign(:value, assigns[:value] || field.value)
...
```

with this we don't have to remove default value for `name` and `value` attributes